src/pipe.js

```diff
@@ -1,34 +1,31 @@
-var compose = require('./compose');
-var reverse = require('./reverse');
+var _pipe = require('./internal/_pipe');
+var curryN = require('./curryN');
+var reduce = require('./reduce');
+var tail = require('./tail');
 
 
 /**
- * Creates a new function that runs each of the functions supplied as parameters in turn,
- * passing the return value of each function invocation to the next function invocation,
- * beginning with whatever arguments were passed to the initial invocation.
- *
- * `pipe` is the mirror version of `compose`. `pipe` is left-associative, which means that
- * each of the functions provided is executed in order from left to right.
+ * Performs left-to-right function composition. The leftmost function may have
+ * any arity; the remaining functions must be unary.
  *
  * In some libraries this function is named `sequence`.
+ *
  * @func
  * @memberOf R
  * @category Function
- * @sig ((a... -> b), (b -> c), ..., (x -> y), (y -> z)) -> (a... -> z)
- * @param {...Function} functions A variable number of functions.
- * @return {Function} A new function which represents the result of calling each of the
- *         input `functions`, passing the result of each function call to the next, from
- *         left to right.
+ * @sig ((a -> b), (b -> c), ..., (y -> z)) -> (a -> b -> c -> ... -> y -> z)
```

Comment:
I'm nor sure what would be a reasonable sig for your original proposal.  The change I was noticing was that the resulting function now can have an arity greater than the original (unary-only?) first one, according to this sig.  This is a substantial change from current behavior, and there are no tests for it.

I haven't downloaded this version to play with it as I did the earlier one.  It didn't _look_ like this version actually did that, so I guess my question was a way of asking if I was reading the code correctly.

In any case, I think this sig is wrong.  In any version of `compose` we're likely to have the intermediate types in a string of binary functions (`b`, `c`, and `y` above) shouldn't show up in the resulting function.  

The older sig was somewhat misleading too.  That `a...` wasn't descriptive enough.  This might have been a better version:

    pipe :: ((a -> b -> ... -> n), (n -> o), (o -> p), ..., (x ->y), (y -> z)) -> (a -> b -> ... -> n -> z)